src/dlt_packet_slice.rs

```diff
@@ -537,6 +721,41 @@ impl<'a> DltPacketSlice<'a> {
     }
 }
 
+fn determine_dlt_typed_playload_for_non_verbose_response(
+    non_verbose_payload: &[u8],
+    message_id: u32,
+    message_info: Option<DltMessageInfo>,
+) -> Option<DltTypedPayload<'_>> {
+    if non_verbose_payload.len() > 5 {
+        let control_message = match non_verbose_payload[0] {
```

Comment:
I think the message id should be used here to identify the service id of the control message

```suggestion
        let control_message = match message_id {
```